powerdns_recursor/datadog_checks/powerdns_recursor/powerdns_recursor.py

```diff
@@ -126,6 +126,27 @@ def check(self, instance):
                     self.gauge('powerdns.recursor.{}'.format(stat['name']), float(stat['value']), tags=tags)
                 elif stat['name'] in PowerDNSRecursorCheck.RATE_METRICS_V4:
                     self.rate('powerdns.recursor.{}'.format(stat['name']), float(stat['value']), tags=tags)
+        self._collect_metadata(config)
+
+    def _collect_metadata(self, config):
+        url_v4 = "http://{}:{}/api".format(config.host, config.port)
+        url = "http://{}:{}/servers/localhost/statistics".format(config.host, config.port)
+
+        try:
+            response = self._get_pdns_response(config, url, url_v4)
+        except Exception as e:
+            self.log.warning('Error collecting PowerDNS Recursor version: %s', str(e))
+            return
+
+        if response.headers.get('Server'):
+            try:
+                # 'Server': 'PowerDNS/4.0.9'
+                version = response.headers['Server'].split('/')[1]
+                self.set_metadata('version', version)
+            except Exception as e:
+                self.log.warning('Error while decoding PowerDNS Recursor version: %s', str(e))
+        else:
+            self.log.warning("Couldn't find the PowerDNS Recursor Server version header")
 
     def _get_config(self, instance):
```

Comment:
Side note [not for this PR]: unless required, ideally config parsing should be done in the constructor instead of on each check run (we've seeking to adopt this pattern across integrations, eg see #6924).

So a possible follow up PR could be to move this to `__init__()` and store as `self.config`.